src/hu/openig/model/Planet.java

```diff
@@ -710,6 +710,10 @@ public void takeover(Player newOwner) {
 			lastOwner.statistics.planetsLostAlien.value++;
 		}
 		lastOwner.statistics.planetsLost.value++;
+		
+		// unfinished buildings need to be removed from the surface
+		// before granting technology for captured buildings.
+		removeUnfinishedBuildings();
```

Comment:
I'm not certain this call would be needed outside this method and I'd merge that loop with the tech-capturing loop just below:

```java
for (Building b : this.surface.buildings.list()) {
    if (!b.isComplete()) {
        this.surface.removeBuilding(b);
    } else
    if (b.type.research != null) {
        newOwner.setAvailable(b.type.research);
    }
}
rebuildRoads();
```

Reply:
Good call, less loops == better.